pkg/operator/ceph/controller/predicate.go

```diff
@@ -60,14 +60,14 @@ func WatchControllerPredicate() predicate.Funcs {
 				objNew := e.ObjectNew.(*cephv1.CephObjectStore)
 				logger.Debug("update event from the parent object CephObjectStore")
 				diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
-				if diff != "" ||
-					objOld.GetDeletionTimestamp() != objNew.GetDeletionTimestamp() ||
-					objOld.GetGeneration() != objNew.GetGeneration() {
+				if diff != "" || objOld.GetDeletionTimestamp() != objNew.GetDeletionTimestamp() {
```

Comment:
@leseb Given the fix for #5224 should this check only be looking at the generation? The generation should tell us if there is a diff in the spec or if there is a deletion timestamp, right?  
```
if objOld.GetGeneration() != objNew.GetGeneration() {
...
}
```

Reply:
I don't think we need to reconcile on a generation because IIRC a generation is an updated version of let's say a deployment so based on that the object is already up to date. This particular code is not triggered on CR update but only on child resources like dep, secret, srv etc.

Reply:
Every resource has a generation (int64) that is guaranteed to be incremented every time the resource changes. Since we're comparing the generation of the previous version of the resource with the new one, we will know if it changed without checking the diff. The generation must be changing with status updates as well, but with the fix in #5224 this event must not be called so it still seems like we could rely on the generation without checking the diff. Need to confirm though...